diff --git a/Dockerfile b/Dockerfile
index 31a6229..32cd227 100755
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,6 +1,8 @@
 FROM ghcr.io/bayeswatch/bwatch-tutorial:latest
 
 RUN apt update
+RUN apt upgrade -y
+
 RUN apt install aptitude tree -y
 RUN apt install fish -y
 RUN echo y | pip install tabulate nvitop hydra_zen
@@ -8,7 +10,6 @@ RUN mamba install -c conda-forge starship jupyterlab black -y
 
 RUN git lfs install
 RUN git config --global credential.helper store
-RUN git config --global --add safe.directory /workspaces/minimal-ml-template
 
 RUN echo y | pip uninstall torch
 RUN mamba uninstall pytorch -y
@@ -16,4 +17,10 @@ RUN mamba uninstall pytorch -y
 RUN mamba install pytorch torchvision torchaudio pytorch-cuda=11.6 -c pytorch -c nvidia -y
 RUN mamba install -c conda-forge timm accelerate datasets transformers -y
 
-ENTRYPOINT ["/bin/bash"]
+COPY . /workspace/minimal-ml-template/
+RUN git config --global --add safe.directory /workspace/minimal-ml-template
+
+RUN apt install kubectl -y
+RUN apt install google-cloud-sdk-gke-gcloud-auth-plugin -y
+
+ENTRYPOINT ["/bin/bash", "/workspace/minimal-ml-template/entrypoint.sh"]
diff --git a/mlproject/config.py b/mlproject/config.py
index 848ee8e..bf57630 100755
--- a/mlproject/config.py
+++ b/mlproject/config.py
@@ -102,8 +102,7 @@ class BaseConfig:
     # Must be passed at command line -- neccesary arguments
 
     exp_name: str = MISSING
-    hf_username: str = MISSING
-
+    
     # Defaults for these are provided in the collect_config_store method,
     # but will be often overridden at command line
 
@@ -116,6 +115,11 @@ class BaseConfig:
     callbacks: Any = MISSING
 
     wandb_args: Any = MISSING
+    
+    hf_username: str = (
+        os.environ["HF_USERNAME"] if "HF_USERNAME" in os.environ else MISSING
+    )
+
 
     seed: int = 42
 
@@ -140,6 +144,8 @@ class BaseConfig:
     data_dir: str = (
         os.environ["DATASET_DIR"] if "DATASET_DIR" in os.environ else "/data"
     )
+    
+
 
     current_experiment_dir: str = "${root_experiment_dir}/${exp_name}"
     repo_path: str = "${hf_username}/${exp_name}"
diff --git a/mlproject/run.py b/mlproject/run.py
index 0b467b9..43b35de 100755
--- a/mlproject/run.py
+++ b/mlproject/run.py
@@ -229,6 +229,8 @@ def run(cfg: BaseConfig) -> None:
 
     config_dict = OmegaConf.to_container(cfg, resolve=True)
     wandb_args["config"] = config_dict
+
+    wandb.login(key=os.getenv("WANDB_API_KEY"))
     wandb.init(**wandb_args)  # init wandb and log config
 
     upload_code_to_wandb(cfg.code_dir)  # log code to wandb
